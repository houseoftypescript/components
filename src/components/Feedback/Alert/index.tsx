import React, { ReactNode } from 'react';
import styled from 'styled-components';

type AlertType = 'error' | 'success' | 'warning' | 'info';

const colors: Record<AlertType, string> = {
  error: '#ef4444',
  success: '#22c55e',
  warning: '#eab308',
  info: '#3b82f6',
};

const AlertContainer = styled.div<{
  type: AlertType;
}>`
  display: flex;
  border-radius: 0.25rem;
  overflow: hidden;
  gap: 0.5rem;
  padding: 0.5rem 1rem;
  background-color: ${(props) => colors[props.type]};
  color: #ffffff;
`;

const AlertTitle = styled.div`
  font-weight: 700;
`;

const AlertDescription = styled.div``;

const Alert: React.FC<{
  type: AlertType;
  children?: ReactNode;
}> = ({ type = 'info', children = <></> }) => {
  return <AlertContainer type={type}>{children}</AlertContainer>;
};

export default Object.assign(Alert, {
  Title: AlertTitle,
  Description: AlertDescription,
});
