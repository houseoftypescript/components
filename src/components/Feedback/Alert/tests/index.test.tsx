import { render } from '@testing-library/react';
import React from 'react';
import Alert from '../index';

describe('Alert', () => {
  it('to match snapshot', () => {
    const { container } = render(
      <Alert type="info">
        <Alert.Title>Title</Alert.Title>
        <Alert.Description>Description</Alert.Description>
      </Alert>
    );
    expect(container).toMatchSnapshot();
  });
});
