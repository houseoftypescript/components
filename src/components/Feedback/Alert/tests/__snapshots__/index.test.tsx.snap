// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert to match snapshot 1`] = `
<div>
  <div
    class="sc-beySbM HOEaZ"
    type="info"
  >
    <div
      class="sc-guDLRT imaiJM"
    >
      Title
    </div>
    <div
      class="sc-dmyDGi"
    >
      Description
    </div>
  </div>
</div>
`;
